### ISSUE TITLE
Unclear how and where random number seeds are used in CT

### ISSUE THREAD

**Issue by rickdonnelly** [pbsag/swimctr#4]
Sampling is used extensively within the CT functions, but random number seeds are not explicitly set in some cases. They need to be handled in formal and consistent way in all of the functions. This is important to ensure reproducibility of results, as well as consistency across different model components.

**Comment by rickdonnelly** [comment]
In order to obtain some semblance of reproducibility I've started using compound seeds that are concatenation of run number and the entity ID. For example, if I'm generating daily trucks by firm the entity ID would be the firm ID, and we'd reset the random seed for each new firm. The rub in this case is that we don’t assign unique run number or scenario ID when running SWIM. So I could use zero, or the day (in yyddd format), as the default prefix. The latter might be implemented as:

```{
today <- format(Sys.Date(), "%y%j")  # Would return 16342 on 7-Dec-2016
random_seed <- as.integer(paste(today, firmID, sep = ''))
```

This code should be run each time we process a new entity (e.g., firm, truck, shipment). That way I can slice the problem into different chunk sizes in [doParallel](https://cran.r-project.org/web/packages/doParallel/index.html) without worrying about the results being different because the order of execution has changed. 

It is my intention to eventually code the prefix as random seed used in the antithetical case, based upon running CT hundreds of times, varying them only by the random seed. I'll then find the run that produced the least amount of variation and smallest number of outliers across several metrics, and use its prefix as the default. This will push the stochastic variation to the vanishing point is we run the model hundreds of times, leaving only inherent behavioral variability left in the results. However, I don't believe we should hold up addressing this issue in code until we complete that task.

My proposal is to create a function parameter named `random_seed_prefix` that is included in all function headers within swimctr that uses sampling. By default it will be set to the SWIM runtime parameter `ct.random.seed` if that exists, and zero otherwise. Since this will be used across several functions we'll include a `set_random_seed()` function in helpers.R to ensure consistency across different parts of the model.

**Comment by rickdonnelly** [comment]
After testing several variants on this I've found out how hard it is to actually control a random number seed that lasts through several invocations of doParallel. Moreover, in the grand scheme of things the outcomes varied very little -- enough to measure, not enough to care about. I think I've reached the vanishing point on this, accepting that there are _many_ more important things to pay attention to than this.
